packages/scandipwa/src/component/ProductBundleItems/ProductBundleItems.container.js

```diff
@@ -73,42 +78,82 @@ export class ProductBundleItemsContainer extends ProductCustomizableOptionsConta
         this.setState({ isLoading: false });
     }
 
-    getOptionPrice(item, selectedValues) {
+    getDynanicOptionPrice(product, quantity) {
+        const {
+            price_range: {
+                minimum_price: {
+                    regular_price: {
+                        value: itemPrice = 0
+                    } = {},
+                    final_price: {
+                        value: finalItemPrice = 0
+                    } = {},
+                    final_price_excl_tax: {
+                        value: finalItemPriceExclTax = 0
+                    } = {}
+                } = {}
+            } = {}
+        } = product;
+
+        const optionInitialPrice = itemPrice * quantity;
+        const optionPrice = finalItemPrice * quantity;
+        const optionPriceExclTax = finalItemPriceExclTax * quantity;
+        return {
+            optionPrice,
+            optionPriceExclTax,
+            optionInitialPrice
+        };
+    }
+
+    getOptionPrice(item, selectedValues, bundleDefaultPrices, isDynamicPrice) {
         const { option_id } = item;
         let price = 0;
         let priceExclTax = 0;
+        let initialPrice = 0;
```

Comment:
This function is too long. Please split it in parts to improve extensibility.
Usually when devs extend components, they are copying the whole method in order to change some lines.
The more atomic methods are, the better. 

Also try to avoid use of let. This should be forbidden by eslint configuration.
It may be disabled now per file or globally. But still we should avoid it in order to prevent us from fixing this code when it will be enabled again.

Reply:
I get your point. But current method includes a 2-level foreach loop + quite a lot of logic. I can make the replacement, but I'm afraid that if I use `reduce` instead of `let`, the code will get less readable.